### DISCUSSION TITLE
JWT 이름을 클라이언트에 어떻게 저장하는 것이 좋을까?

### DISCUSSION THREAD

**Discussion by beatmeJY** [dailyge/dailyge-server#136] [Discussion]
## 🔗 연관 PR

- [[DLG-209] AccessToken을 Cookie에 저장하도록 수정한다.](https://github.com/dailyge/dailyge-server/pull/126)

<br/><br/>

## ❓ 작성 이유
이전에 @devjun10님과 JWT를 저장 시 이름을 잘 알아보지 못하도록 변경하여 올려두자고 이야기를 한 적이 있었는데요.

따라서 저희 팀은 어떠한 이름을 사용하면 좋을 지에 대하여 의견을 나누기 위해 작성하였습니다.

<br/><br/>

## 📝 내용
> 먼저 유명한 웹사이트들에서는 어떻게 사용하는지 알아보겠습니다.

<br/>

### 네이버
![image](https://github.com/user-attachments/assets/d94e84a4-6023-4656-9412-0121ae6997c3)
- 확실하게는 알 수 없지만 `NID_AUT`가 Naver ID_Auth를 의미하고 `NID_SES`는 Naver ID_Session을 뜻한다고 생각할 수 있을 것 같습니다.
<br/><br/>

### 구글
![image](https://github.com/user-attachments/assets/559efaa2-cac2-4cdd-8cc2-47f77a6b228a)
 - 위 사진을 보면 다양한 쿠키와 비슷한 이름들이 많아서 어떤 값이 인증을 위한 쿠키인지 확신할 수 없지만
 - 아래의 사진을 보면 인증을 위해서 `SID`와 `HSID`를 사용한다고 볼 수 있습니다.

![image](https://github.com/user-attachments/assets/e56b80a2-691c-4c78-a2fb-f3f93c2daa0e)

<br/><br/>
### ChatGPT
![image](https://github.com/user-attachments/assets/72fc43fe-0f2d-415e-a942-36d7cbf781d2)
 - ChatGPT는 token이라는 명확한 이름으로 `_Secure-next-auth.session-token` 란 쿠키 명을 사용하는 것을 확인해볼 수 있었습니다.

<br/><br/>

### GitHub
![image](https://github.com/user-attachments/assets/9f03f300-7632-4b31-bf6c-c861dee815af)
 - 깃허브는 `_gh_sess` 로 GitHub Session 이라는 쿠키명을 사용하는 것으로 유추해볼 수 있었습니다.

<br/><br/><br/><br/>

따라서 제 결론으로는 완전히 알아보기 힘들게 쿠키의 이름을 설정하지는 않고 있는 것 같아 정리를 해보자면,
- `DID`
- `dg_sess`

저는 위와 같은 쿠키명을 사용해도 좋을 것 같다 생각하였는데, 팀원분들의 의견도 댓글로 달아주시면 감사하겠습니다!😄

### 결론
회의 결과 아래와 같이 사용하도록 결정되었습니다.
 - 엑세스 토큰 `dg_sess`
 - 리프레시 토큰 `dg_res`

**Comment by devjun10** [comment]
저는 세션이란 단어가 들어가면 좋을 것 같네요. 이거 다음 주 회의 때, 한 번 같이 이야기해보죠.

**Comment by kmularise** [comment]
저도 세션 들어가면 좋을 거 같아요! 저도 실제 서비스에서 어떻게 쿠키가 사용되는지 들여다봐야겠네요.
